### ISSUE TITLE
Android integration

### ISSUE THREAD

**Issue by PopTudor** [TadasBaltrusaitis/OpenFace#9]
Hello !
I was thinking to integrate OpenFace with Android but I saw it uses about 100mb ram on ubuntu and 200 mb ram on windows which is too much for a device but seeing this difference in os I'm thinking weather it will work on Android where 50/100 mb is about the maximum you can get. Do you think it can handle it ? Also, do you have any suggestions on how one should proceed with the integration ? I'm not sure if I should generate .a or .so  files and if it will work out of the box like this.

**Comment by PopTudor** [comment]
I've seen others have asked the same question on the old repo. I will give it a try and if successful, I will reply here

**Comment by deepakakkil** [comment]
Thanks for providing this software as opensource. Do you have any updates on an android or ios version?

**Comment by PopTudor** [comment]
@deepakakkil not yet, I only had about 2 days to try, then my schedule was full. I will try more this and the following week

**Comment by makerGeek** [comment]
any updates?

**Comment by PopTudor** [comment]
nope. I found it difficult to add the 3rd party libraries (boost etc...) to android. It can be done but I literally have 0 experience with the NDK and this library.

**Comment by vanhoanHoang** [comment]
Do you have any updates ? Tks.

**Comment by eladrich** [comment]
I was wondering, did anyone got an Android port working? If not, I'm thinking on starting one myself.

@TadasBaltrusaitis, could you tell me how dependent is OpenFace on the 3rd party libraries (boost, dlib, tbb)?
While a stable OpenCV port exists for Android, the other could be challenging to port. So it might be easier to just get rid of these dependencies completely.

**Comment by TadasBaltrusaitis** [comment]
dlib is used for face detection and to help Facial Action Unit detection by extracting Histogram of Oriented Gradients features
boost is mainly used for directory and file traversal in OS independent way
tbb is used to speed up computation by parallelising it

It will not be too difficult to get rid of boost or tbb (but will require a reasonable amount of coding), replacing dlib will be much more challenging but not impossible.

Thanks,
Tadas

**Comment by meansoup** [comment]
Did anyone gone a step for android?
I need to find landmark at android. If anyone done, I want to do that, but have less time.

**Comment by eladrich** [comment]
I was able to get a basic version working on Android, based on the [iOS port](https://github.com/FaceAR/OpenFaceIOS), but it's not very clean right now.
Basically what I've done is to remove the boost dependencies, by matching the file traversal to Android, and get it to work with existing dlib and openCV ports.

Hopefully, I'll find some time to clean it and put it on Github.

**Comment by meansoup** [comment]
@eladrich thanks for your reply.
I'm wandering now for using this, I just want to use face landmark (at img or at video)
so anaylze the code, and find that FaceLandmarkVidMulti.cpp did not use boost or tbb, just use dlib.(I think we can use dlib. isn't it?)
so, I add all the files which I need to find landmark to my android project. but... now i don't know how to do.
My thinking is that make java mainactivity like FaceLandmarkVidMulti.cpp, and use c++ method via cmake.
theorically, it can be? can you give me an advice??
thanks.

**Comment by eladrich** [comment]
@meansoup if you just need face landmarks, take a look at this dlib port:
https://github.com/tzutalin/dlib-android

**Comment by meansoup** [comment]
@eladrich thanks.

Perfectly what I was looking for is that. and I have to done more analyse.

thanks very much

**Comment by JLeon3** [comment]
Has anyone been successful with getting it to work with Android app?

**Comment by gicheonkang** [comment]
Hi, @JLeon3 
I made a app with detecting 68 landmarks. Not Openface version..
here is my git repo. https://github.com/gicheonkang/Fast-Face
It just takes 60ms to detecting your landmark points. Enjoy it :)

**Comment by idireneyoucef** [comment]
hi, im new in android and i want to create  an application  for  matching  face like on facebook  how i can begin  and openFace  it's the best solution ? help please

**Comment by TadasBaltrusaitis** [comment]
OpenFace does not have an Android version, and is not directly intended for face recognition, but rather facial behavior analysis.

**Comment by idireneyoucef** [comment]
thanx for answered,
do You have an idea of api of facial recognition?

**Comment by Q-lin** [comment]
@eladrich Has you successful with getting it to work with Android app?please help me
thanks very much

**Comment by amogh112** [comment]
@eladrich can you please share your Android implementation like you have mentioned above?

**Comment by R00iBaard** [comment]
> I was able to get a basic version working on Android, based on the [iOS port](https://github.com/FaceAR/OpenFaceIOS), but it's not very clean right now.
> Basically what I've done is to remove the boost dependencies, by matching the file traversal to Android, and get it to work with existing dlib and openCV ports.
> 
> Hopefully, I'll find some time to clean it and put it on Github.

@eladrich did you ever come right with this? Would it be possible to contact you privately about this even?
